docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md

```diff
@@ -205,6 +225,8 @@ Below are described the steps you need to make to migrate from v6 to v7.
   | `unstable_gridHeaderFilteringStateSelector`       | `gridHeaderFilteringStateSelector`       |
   | `unstable_gridTabIndexColumnHeaderFilterSelector` | `gridTabIndexColumnHeaderFilterSelector` |
 
+- The filter panel no longer uses the native version of the [`Select`](https://mui.com/material-ui/react-select/) component for all components.
```

Comment:
A small regression on header filters keyboard navigation due to SelectInput component [intercepting Enter key](https://github.com/mui/material-ui/blob/a5d6da7490c9dcf59707df43ed12725b63e933c7/packages/mui-material/src/Select/SelectInput.js#L315-L317) which results in focus not shifting back to the header cell properly. IIRC for the same reason the keyboard navigation doesn't work properly for Aggregation item in the column menu.

Prev and After:
![hf-prev](https://github.com/mui/mui-x/assets/12609561/4c783e11-3219-400c-b6f7-c4abc4463f88) ![hf-after](https://github.com/mui/mui-x/assets/12609561/87e2e149-f66e-480b-afb1-ee308e2da172)

I am investigating on how to solve this, do you think falling back to the native select in header filters and aggregation item could be a viable option for now?

Reply:
Oh, I didn't check the header filters...
What would be the expected behavior though? Since it's recommended for non-native select to open on Enter (and I honestly expected this as well), would it work if we don't stop edit mode in that case?
```diff
diff --git a/packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterCell.tsx b/packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterCell.tsx
index 5cb84a07d..b7b6c3c36 100644
--- a/packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterCell.tsx
+++ b/packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterCell.tsx
@@ -141,8 +141,10 @@ const GridHeaderFilterCell = React.forwardRef<HTMLDivElement, GridHeaderFilterCe
             break;
           case 'Enter':
             if (isEditing) {
-              apiRef.current.stopHeaderFilterEditMode();
-              break;
+              if (!event.defaultPrevented) {
+                apiRef.current.stopHeaderFilterEditMode();
+                break;
+              }
             }
             if (event.metaKey || event.ctrlKey) {
               headerFilterMenuRef.current = buttonRef.current;

```

Reply:
It's better now that it's not stuck on no point of return, but there's still a small issue; once the user is focused on a **Select**, there's no way to focus back to the cell and (or) navigate to another cell using the keyboard. After #4658 is implemented though, it should be fixed as the keyboard navigation will be possible in this case using the <kbd>Tab</kbd> key.
We can treat this as a corner case for now which will be automatically addressed by #4658.
Wdyt?

Reply:
> once the user is focused on a Select, there's no way to focus back to the cell and (or) navigate to another cell using the keyboard.

Yes, Enter doesn't exit the edit mode, but Escape works.
I think this is acceptable for now.